### PR TITLE
loader support alternative and sourcedb options

### LINKED ISSUE #137: loader: Requiring to support -B, -s and -C options of myloader
Requirements: If loader could support -B, -s and -C options of myloader, it will be convenient for users.

  -B, --database                    An alternative database to restore into
  -s, --source-db                   Database to restore
  -C, --compress-protocol           Use compression on the MySQL connection

### PR DESCRIPTION
**Pull request by jomenxiao**
support alternative/source options

fix #137

### CODE REVIEW

**Comment by jomenxiao** [comment]
@holys @GregoryIan  PTAL

**Comment by GregoryIan** [comment]
I think it's not ready for reviewing

**Comment by jomenxiao** [comment]
LGTM

File: loader/config.go [hunk 1]
```diff
@@ -131,16 +136,39 @@ func (c *Config) Parse(arguments []string) error {
 		return errors.Errorf("'%s' is an invalid flag", c.FlagSet.Arg(0))
 	}
 
+	if (c.AlternativeDb != "") == (c.SourceDb != "") {
```
Comment (holys) [inline loader/config.go]: should be `if (c.AlternativeDb != "") != (c.SourceDb != "")`

Comment (GregoryIan) [inline loader/config.go]: lol, same mistake. I must miss `==` very much while I write `!=`

File: loader/config.go [hunk 2]
```diff
@@ -131,16 +136,39 @@ func (c *Config) Parse(arguments []string) error {
...
+		rule := &RouteRule{
+			PatternSchema: strings.ToLower(c.SourceDb),
+			TargetSchema:  c.AlternativeDb,
+		}
+		routeRules = append(routeRules, rule)
+	}
+
 	for _, rule := range c.RouteRules {
+		if c.AlternativeDb != "" {
+			if rule.TargetSchema == c.SourceDb {
```
Comment (holys) [inline loader/config.go]: Why compare here ` rule.TargetSchema == c.SourceDb` ?

**Comment by holys** [comment]
LGTM

**Pull request merged**